File: algorithms/code/leetcode/lc011_container_with_most_water/LC011ContainerWithMostWater.java
package leetcode.lc011_container_with_most_water;

/**
 * https://leetcode.com/problems/container-with-most-water/
 */
public final class LC011ContainerWithMostWater {
    public int maxArea(final int[] height) {
        int max = 0;
        int low = 0;
        int high = height.length - 1;
        while (low < high) {
            int top = Math.min(height[low], height[high]);
            max = Math.max(max, top * (high - low));
            if (height[low] <= height[high]) {
                low++;
            } else {
                high--;
            }
        }
        return max;
    }
}

File: algorithms/code/leetcode/lc011_container_with_most_water/lc011_container_with_most_water.go
// Package lc011 implements https://leetcode.com/problems/container-with-most-water/
package lc011

import (
	"math"
)

func maxArea(height []int) int {
	maximum := 0
	low := 0
	high := len(height) - 1
	for low < high {
		top := int(math.Min(float64(height[low]), float64(height[high])))
		maximum = int(math.Max(float64(maximum), float64(top*(high-low))))
		if height[low] <= height[high] {
			low++
		} else {
			high--
		}
	}
	return maximum
}
